test/test_server.c

```diff
@@ -150,6 +169,27 @@ int main_server(void)
 			exit(-1);
 		}
 
+		err = gnutls_record_get_state(session, 0, &mac_key, &iv,
+					      &cipher_key, seq_number);
+		if (err) {
+			gnutls_perror(err);
+			exit(-1);
+		}
+
+		memcpy(crypto_info.iv, iv.data + 4,
```

Comment:
Do we really need to redo these?

Why we can't go straight for this?
```C
		if (setsockopt(client, SOL_TLS, TLS_TX, &crypto_info,
			       sizeof(crypto_info))) {
			perror("setsockopt[tls]");
			exit(-1);
		}
```

Or even better:
```C
if (setsockopt(client, SOL_TLS, TLS_RX|TLS_TX, &crypto_info,
```
It seems that it's the exact same information.

Reply:
according to `gnutls_record_get_state` document here https://www.gnutls.org/manual/gnutls.html
The read and write param maybe different controlled by `read` param, so I think we must split the RX and TX